File: android/src/main/java/dev/barooni/capacitor/calendar/models/results/CreateCalendarResult.kt
package dev.barooni.capacitor.calendar.models.results

import com.getcapacitor.JSObject
import dev.barooni.capacitor.calendar.PluginError
import dev.barooni.capacitor.calendar.models.templates.JSResult

data class CreateCalendarResult(
    private val id: String?,
) : JSResult {
    init {
        if (id == null) {
            throw PluginError.FailedToRetrieveCalendarId
        }
    }

    override fun toJSON(): JSObject {
        val result = JSObject()
        result.put("id", id)
        return result
    }
}

File: ios/Sources/CapacitorCalendarPlugin/Models/Results/CreateCalendarResult.swift
import Capacitor

struct CreateCalendarResult: JSResult {
    let id: String

    init(id: String?) throws {
        guard let id = id else {
            throw PluginError.failedToRetrieveCalendarId
        }
        self.id = id
    }

    func toJSON() -> JSObject {
        var result = JSObject()
        result["id"] = id
        return result
    }
}
